run_distillation.py

```
@@ -213,467 +232,603 @@ def parse_args_to_config() -> TrainingConfig:
+
+        if self.is_main_rank:
+            free_summary = ", ".join(f"{idx}:{gpu_free_mem[idx][0]:.1f}GB" for idx in local_avail)
+            print(f"Eligible GPUs (> {min_free_gb:.0f}GB free): [{free_summary}]", flush=True)
+            print(
+                f"[device-planning] Using single GPU cuda:{chosen_gpu} for both teacher and student.",
+                flush=True,
+            )
+
+        if self.config.ddp_offline:
+            if self.ddp_local_rank >= len(self.local_avail):
+                raise RuntimeError(f"LOCAL_RANK={self.ddp_local_rank} exceeds available GPUs {self.local_avail}")
```

Ruff (0.14.1)

Avoid specifying long messages outside the exception class

(TRY003)